pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -256,6 +384,201 @@ func (r *repository) Delete(dgst digest.Digest) error {
 	return ms.Delete(dgst)
 }
 
+// Stat makes a local check for the blob, then falls through to the other servers referenced by
+// the image stream and looks for those that have the layer.
+func (r *blobs) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
+	// check the local store for the blob
+	desc, err := r.BlobStore.Stat(ctx, dgst)
+	if err == nil || err != distribution.ErrBlobUnknown {
```

Comment:
Shouldn't this be
```go
if err != nil && err != distribution.ErrBlobUnknown {
```
Also the error message here could be more explicit than `Failed to find blob` (a failed search - `ErrBlobUnknown` also implies a failed search).